craft_parts/sources/local_source.py

```diff
@@ -43,10 +44,10 @@ class LocalSource(SourceHandler):
 
     def __init__(
         self,
-        *args: Any,  # noqa: ANN401
+        *args: tuple[Any, ...],
         project_dirs: ProjectDirs,
         copy_function: Callable[..., None] = file_utils.link_or_copy,
-        **kwargs: Any,  # noqa: ANN401
+        **kwargs: dict[str, Any],
```

Comment:
I'm a bit surprised that this works, I thought for kwargs you only had to annotate the value type (since the key type is always `str`). So this would mean that `kwargs` is a `dict[str, dict[str, Any]]`
but well, if it works it works ;)

Reply:
After check PEP, the correct typing for this are `*args: Any` and `**kwargs: Any`.
Since `*args: tuple(T, ...)` and `**kwargs: dict[str, T]`, if we don't what's in it, it has to be `Any`.
@cmatsuoka